### PR TITLE
Issue 16  doc warnings

### PR DESCRIPTION
**Pull request by adler-j**
Fixed most doc warnings so far, still a few to go. Need review and tests with `cuda` part. Also needs to migrate the changes to `solvers` to `odl-solvers`.

### CODE REVIEW

**Comment by adler-j** [comment]
Added some more commits, still a warning in `ProductSpace.__init__` that simply *wont* go away, I cannot find it. Need some help.

**Comment by adler-j** [comment]
Other than that we only have 3 warnings about objects not beeing included in the toctree, but that should be solvable.

**Comment by kohr-h** [comment]
I get a few more warnings, some of which may be related to outdated packages. An obvious one is the `Yields` section in `numpydoc`, supported since version 0.6, but that has not even hit PyPI.
`Usage` may be in some utility function, maybe progress bar? That should obviously be solved otherwise. I'll try to fix the product space problem.

Here's a dump:
```
[hkohr@cormack doc]$ make html
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v1.2.3
loading pickled environment... done
loading intersphinx inventory from http://python.readthedocs.org/en/latest/objects.inv...
loading intersphinx inventory from http://docs.scipy.org/doc/scipy/reference/objects.inv...
loading intersphinx inventory from http://numpy.readthedocs.org/en/latest/objects.inv...
loading intersphinx inventory from http://matplotlib.sourceforge.net/objects.inv...
building [html]: targets for 11 source files that are out of date
updating environment: [extensions changed] 26 added, 0 changed, 5 removed
/usr/lib/python2.7/site-packages/numpydoc/docscrape.py:120: UserWarning: Unknown section Yields                                                                                               
  warn("Unknown section %s" % key)
isinstance() arg 2 must be a class, type, or tuple of classes and types                                                                                                                       
/usr/lib/python2.7/site-packages/numpydoc/docscrape.py:120: UserWarning: Unknown section Usage                                                                                                
  warn("Unknown section %s" % key)

/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/dev.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/configure_git.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/development_workflow.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/following_latest.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/forking_hell.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/git_development.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/git_install.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/git_intro.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/git_resources.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/index.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/maintainer_workflow.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/patching.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/gitwash/set_up_fork.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/guide/guide.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/guide/introduction/about_odl.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/guide/introduction/getting_started.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/guide/introduction/index.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/guide/introduction/installing.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/index.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/odl.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/odl.rst:19: WARNING: missing attribute mentioned in :members: or __all__: module odl, attribute sets
/home/hkohr/Work/Projects/2015/LCR/odl/odl/__init__.py:docstring of odl.ProductSpace.__init__:59: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/home/hkohr/Work/Projects/2015/LCR/odl/doc/odl.diagnostics.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/odl.discr.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/odl.operator.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/odl.set.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/odl/set/pspace.py:docstring of odl.set.pspace.ProductSpace.__init__:59: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/home/hkohr/Work/Projects/2015/LCR/odl/odl/set/__init__.py:docstring of odl.set.ProductSpace.__init__:59: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/home/hkohr/Work/Projects/2015/LCR/odl/doc/odl.space.rst:: WARNING: default role any not found
/home/hkohr/Work/Projects/2015/LCR/odl/doc/odl.space.rst:18: WARNING: autodoc: failed to import module u'odl.space.cu_ntuples'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
  File "/home/hkohr/Work/Projects/2015/LCR/odl/odl/space/cu_ntuples.py", line 1195, in <module>
    raise ImportError('Your GPU seems to be misconfigured. Skipping '
ImportError: Your GPU seems to be misconfigured. Skipping CUDA-dependent modules.
/home/hkohr/Work/Projects/2015/LCR/odl/doc/odl.util.rst:: WARNING: default role any not found
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/hkohr/Work/Projects/2015/LCR/odl/doc/dev/dev.rst:: WARNING: document isn't included in any toctree
/home/hkohr/Work/Projects/2015/LCR/odl/doc/guide/guide.rst:: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] odl.util                                                                                                                                                             
writing additional files... (25 module code pages) _modules/index genindex py-modindex search
copying images... [100%] dev/gitwash/forking_button.png                                                                                                                                       
copying static files... WARNING: html_static_path entry u'/home/hkohr/Work/Projects/2015/LCR/odl/doc/_static' does not exist
done
copying extra files... done
dumping search index... done
dumping object inventory... done
build succeeded, 34 warnings.

Build finished. The HTML pages are in _build/html.
```

**Comment by kohr-h** [comment]
The `ProductSpace` error is due to the crappy old doc (needs update!!)
```
        args : None
            Create a new vector from scratch.

        The second is to wrap existing vectors:

        args : tuple of :class:`LinearSpace.Vector`
            A tuple of vectors in the underlying spaces.
            This will simply wrap the Vectors (not copy).

        The third pattern is to create a new Vector from scratch, in
        this case

        args : tuple of array-like objects
```
If you comment out that part, the warning goes away.

**Comment by adler-j** [comment]
Please commit fixes to any warnings you get, easier that way.

**Comment by kohr-h** [comment]
> Please commit fixes to any warnings you get, easier that way.

Huh? What did I do if not commit a fix to the warning?

**Comment by adler-j** [comment]
Branch is up to date with master, some fixes done so tests run aswell. Now only has two warnings, but I cannot seem to get em away, if you want to try please do but they dont seem major.

File: doc/source/conf.py [hunk 1]
```diff
@@ -60,10 +69,10 @@ def __getattr__(cls, name):
 ]
 
 #Intersphinx to get numpy targets
-intersphinx_mapping = {'python': ('http://docs.python.org/2', None),
-                       'numpy': ('http://docs.scipy.org/doc/numpy/', None),
-                       'scipy': ('http://docs.scipy.org/doc/scipy/reference/', None),
-                       'matplotlib': ('http://matplotlib.sourceforge.net/', None)}
+intersphinx_mapping = {'python': ('http://python.readthedocs.org/en/latest/', None),
+					   'numpy': ('http://numpy.readthedocs.org/en/latest/', None),
```
Comment (kohr-h) [inline doc/source/conf.py]: PEP8 checker for <your editor>? Would scream here.

Comment (adler-j) [inline doc/source/conf.py]: To many spaces? Dunno why this happened.

Comment (kohr-h) [inline doc/source/conf.py]: Or tabs maybe. Comments without space after `#` would also trigger :wink:

**Comment by kohr-h** [comment]
Down from 3-4 (using `make clean` before the first run) to basically one. One warning was about the `solvers` module it had a hard time finding. One was about the `Usage` section used in the doc of some utility functions. I fixed those two.

The makefile also complains about `source/_static` not existing, but I don't know why it should. Maybe one can just add a command creating the directory in the `Makefile` to shut the warning up.
Now only this warning remains:
```
doc/source/odl.rst:19: WARNING: missing attribute mentioned in :members: or __all__: module odl, attribute sets
```
Not sure what that means. Any idea?

**Comment by adler-j** [comment]
Regarding `_static` we can probably remove it from `conf.py`. The final one I just found out! We had `'sets'` in `odl.__all__`, this should be `'set'`. :)

**Comment by kohr-h** [comment]
Great! Then it's time to merge. (Style can be fixed on master, this stuff is really important to merge)

**Pull request merged**